### PR TITLE
#56 Implement Jenkins

### LINKED ISSUE #56: Implement Jenkins

### PR DESCRIPTION
**Pull request by ilisau**
In this PR I implemented Jenkins for CI/CD and close #56

### CODE REVIEW

**Comment by ilisau** [comment]
I have added Jenkinsfile and Jenkins with docker pods to cluster
Jenkins is also needed to be configured to use docker from another pod

File: .github/workflows/runner.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+name: Auto Build/Deploy
+
+on:
+  push:
+    branches:
+      - "*"
```
Comment (h1alexbel) [inline .github/workflows/runner.yml:6]: should be triggered only on `master` branch

In the case you simple pod listing you can do it on each `pull request` submitted to the `master` branch with your staging environment

File: src/infra/docker-deployment.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+      app: docker
+  template:
+    metadata:
+      labels:
+        app: docker
+    spec:
+      serviceAccountName: jenkins-admin
+      containers:
+        - name: docker
+          image: docker:latest
```
Comment (h1alexbel) [inline src/infra/docker-deployment.yml:19]: use fixed versioning

File: src/infra/runner-serviceaccount.yml [hunk 3]
```diff
@@ -0,0 +1,21 @@
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  namespace: actions-runner-system
+  name: github-runner-api-sa
+automountServiceAccountToken: true
+---
```
Comment (h1alexbel) [inline src/infra/runner-serviceaccount.yml:7]: split this config into multiple files

**Review by h1alexbel — CHANGES_REQUESTED** [review]

**Review by h1alexbel — APPROVED** [review]

**Pull request merged**